nym-registration-client/src/lp_client/nested_session.rs

```diff
@@ -280,6 +270,161 @@ impl NestedLpSession {
         Ok(())
     }
 
+    /// Performs handshake and registration with the exit gateway via forwarding,
+    /// using a pre-made credential.
+    ///
+    /// This variant is useful for mock ecash testing where the credential is provided
+    /// directly instead of being acquired from a bandwidth controller.
+    ///
+    /// # Arguments
+    /// * `outer_client` - Connected LP client with established outer session to entry gateway
+    /// * `wg_keypair` - Client's WireGuard x25519 keypair
+    /// * `credential` - Pre-made bandwidth credential (e.g., mock ecash)
+    /// * `ticket_type` - Type of bandwidth ticket to use
+    /// * `client_ip` - Client IP address for registration metadata
+    ///
+    /// # Returns
+    /// * `Ok(GatewayData)` - Exit gateway configuration data on successful registration
+    pub async fn handshake_and_register_with_credential(
+        &mut self,
+        outer_client: &mut LpRegistrationClient,
+        wg_keypair: &x25519::KeyPair,
+        credential: nym_credentials_interface::CredentialSpendingData,
+        ticket_type: TicketType,
+        client_ip: IpAddr,
+    ) -> Result<GatewayData> {
+        // Step 1: Perform handshake with exit gateway via forwarding
+        self.perform_handshake(outer_client).await?;
+
+        // Step 2: Get the state machine (must exist after successful handshake)
+        let state_machine = self.state_machine.as_mut().ok_or_else(|| {
+            LpClientError::Transport("State machine missing after handshake".to_string())
+        })?;
+
+        tracing::debug!(
+            "Building registration request for exit gateway (with pre-made credential)"
+        );
+
+        // Step 3: Build registration request (credential already provided)
+        let wg_public_key = PeerPublicKey::new(wg_keypair.public_key().to_bytes().into());
+        let request =
+            LpRegistrationRequest::new_dvpn(wg_public_key, credential, ticket_type, client_ip);
+
+        tracing::trace!("Built registration request: {:?}", request);
+
+        // Step 4: Serialize the request
+        let request_bytes = bincode::serialize(&request).map_err(|e| {
```

Comment:
rathen than using the opaque `bincode::serialize`, can you make a helper that explicitly creates `impl bincode::Options` like we do elsewhere in such situations? this is because there are some changes in default behaviour between bincode 1.X and 2.X which would be quite painful to debug once (if) we migrate